src/raz_meta.rs

```diff
@@ -36,27 +36,41 @@ pub trait FirstLast {
 }
 
 impl<E> RazMeta<E> for () {
```

Comment:
If I understand correctly, the logic for this impl should work for _all_ types of meta data, not just meta data of type `()`.  That is, whenever the `Index` type is `OnlyEnds`, the logic below in this impl works.  (Intuitively, all meta data types have the same or strictly more information than `()`).

Perhaps this observation means that `Index` should be a type _parameter_ to the trait `RazMeta`, not an _associated type_; i.e., perhaps this impl should read:

`impl <E,M> RazMeta<E, OnlyEnds> for M { ... }`

Meaning that for all element and meta-data types, `E` and `M`, this impl can index a raz of type `Raz<E,M>` with the index type `OnlyEnds`.